test/Microsoft.Health.Dicom.Tests.Integration/Persistence/WorkitemTests.cs

```diff
@@ -76,5 +81,88 @@ public async Task WhenGetWorkitemQueryTagsIsExecuted_ReturnTagsSuccessfully()
 
             Assert.NotEmpty(workitemQueryTags);
         }
+
+        [Fact]
+        public async Task WhenWorkitemIsQueried_ThenReturnsMatchingWorkitems()
+        {
+            string workitemUid = DicomUID.Generate().UID;
+            DicomTag tag = DicomTag.PatientID;
+
+            var dataset = new DicomDataset();
+            dataset.Add(DicomTag.AffectedSOPInstanceUID, workitemUid);
+            dataset.Add(tag, "FOO");
+
+            var queryTags = new List<QueryTag>()
+            {
+                new QueryTag(new WorkitemQueryTagStoreEntry(2, tag.GetPath(), tag.GetDefaultVR().Code)),
+            };
+
+            long workitemKey = await _fixture.IndexWorkitemStore.AddWorkitemAsync(DefaultPartition.Key, dataset, queryTags, CancellationToken.None);
+
+            var includeField = new QueryIncludeField(new List<DicomTag> { tag });
+            var queryTag = new QueryTag(new WorkitemQueryTagStoreEntry(2, tag.GetPath(), tag.GetDefaultVR().Code));
+            var filters = new List<QueryFilterCondition>()
+            {
+                new StringSingleValueMatchCondition(queryTag, "FOO"),
+            };
+
+            var query = new QueryExpression(QueryResource.WorkitemInstances, includeField, false, 0, 0, filters, Array.Empty<string>());
+
+            var result = await _fixture.IndexWorkitemStore.QueryAsync(DefaultPartition.Key, query, CancellationToken.None);
+
+            Assert.True(result.WorkitemInstances.Any());
+
+            Assert.Equal(workitemKey, result.WorkitemInstances.FirstOrDefault().WorkitemKey);
+        }
+
+        [Fact]
```

Comment:
Could we convert this to a Theory and test a few more cases? Should also demonstrate the behavior if offset or limit are out of range.

Reply:
Added E2e tests to test outofrange offset.